plugins/techdocs/src/api.ts

```diff
@@ -119,14 +119,22 @@ export class TechDocsStorageApi implements TechDocsStorage {
       `${url.endsWith('/') ? url : `${url}/`}index.html`,
     );
 
-    if (request.status === 404) {
-      let errorMessage = 'Page not found. ';
-      // path is empty for the home page of an entity's docs site
-      if (!path) {
-        errorMessage +=
-          'This could be because there is no index.md file in the root of the docs directory of this repository.';
-      }
-      throw new Error(errorMessage);
+    let errorMessage = '';
+    switch (request.status) {
+      case 404:
+        errorMessage = 'Page not found. ';
+        // path is empty for the home page of an entity's docs site
+        if (!path) {
+          errorMessage +=
+            'This could be because there is no index.md file in the root of the docs directory of this repository.';
+        }
+        throw new Error(errorMessage);
+      case 500:
+        errorMessage = 'Could not generate documentation. ';
```

Comment:
Welcome any proposals for other wording here? I suppose it fits my use case, but a 500 error could be anything. I'm referring to a generator message, but if someone is using an external generator, this could also become a preparer or publisher error, so something even _more generic_ but that does not infer "file not found" could be used.

Reply:
We can do "Could not generate documentation or something else went wrong in (backend/server)."

In `TechDocsNotFound` component, there is an additional message based on `techdocsBuilder !== 'local'`, which helps.